xtask/src/build_proto.rs

```diff
@@ -7,18 +7,33 @@ SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
 use clap::Parser;
 
 #[derive(Debug, Parser)]
-pub struct Options {}
+pub struct Options {
+    #[arg(long, default_value_t = false)]
+    controlplane: bool,
+    #[arg(long, default_value_t = false)]
+    dataplane: bool,
+}
 
-pub(crate) fn build_proto(_opts: Options) -> Result<(), anyhow::Error> {
+pub(crate) fn build_proto(opts: Options) -> Result<(), anyhow::Error> {
     let proto_file = "./dataplane/api-server/proto/backends.proto";
 
     println!("building proto {}", proto_file);
 
-    tonic_build::configure()
-        .protoc_arg("--experimental_allow_proto3_optional")
-        .build_server(true)
-        .out_dir("./dataplane/api-server/src")
-        .compile(&[proto_file], &["."])?;
+    if opts.dataplane {
+        tonic_build::configure()
+            .protoc_arg("--experimental_allow_proto3_optional")
+            .build_server(true)
+            .out_dir("./dataplane/api-server/src")
+            .compile(&[proto_file], &["."])?;
+    }
+
+    if opts.controlplane {
+        tonic_build::configure()
+            .protoc_arg("--experimental_allow_proto3_optional")
+            .build_client(true)
+            .out_dir("./controlplane/src/client")
```

Comment:
So there's a slight nuance here that I was thinking about: this is the client for the dataplane, but we're placing it under the controlplane? Would it potentially be better to place this dataplane client under `dataplane/`, even if under it's own crate?

Reply:
Talked it over in Slack: splitting this out into its own PR. :+1: